cwms-data-api/src/main/java/cwms/cda/data/dto/watersupply/WaterUser.java

```diff
@@ -58,13 +58,10 @@ public String getWaterRight() {
     }
 
     @Override
-    public void validate() throws FieldException {
-        if (this.entityName == null || this.entityName.isEmpty()){
-            throw new FieldException("Entity Name cannot be null or empty");
-        }
-        if (this.waterRight == null || this.waterRight.isEmpty()){
-            throw new FieldException("Water Right cannot be null or empty");
-        }
-        this.parentLocationRef.validate();
+    protected void validateInternal(CwmsDTOValidator validator) {
+        super.validateInternal(validator);
+        validator.required(getEntityName(), "entity-name");
```

Comment:
use @JsonProperty(required=true) on the class variable instead of implementing validateInternal